internal/binder/function/funcs_misc.go

```diff
@@ -627,6 +628,123 @@ func registerMiscFunc() {
 			return nil
 		},
 	}
+	builtins["get_keyed_state"] = builtinFunc{
+		fType: ast.FuncTypeScalar,
+		exec: func(ctx api.FunctionContext, args []interface{}) (interface{}, bool) {
+			if len(args) != 3 {
+				return fmt.Errorf("the args must be two or three"), false
+			}
+			key, ok := args[0].(string)
+			if !ok {
+				return fmt.Errorf("key %v is not a string", args[0]), false
+			}
+
+			value, err := keyedstate.GetKeyedState(key)
+			if err != nil {
+				return args[2], true
+			}
+
+			if v, ok := args[1].(string); ok {
```

Comment:
why not use cast package to do the conversion?

Reply:
ack